azure/scope/managedcontrolplane.go

```diff
@@ -210,8 +210,12 @@ func (s *ManagedControlPlaneScope) Close(ctx context.Context) error {
 
 // Vnet returns the cluster Vnet.
 func (s *ManagedControlPlaneScope) Vnet() *infrav1.VnetSpec {
+	vnetResourceGroup := s.ControlPlane.Spec.ResourceGroupName
+	if s.ControlPlane.Spec.VirtualNetwork.ResourceGroup != "" {
```

Comment:
instead of doing this, can we default the value of `s.ControlPlane.Spec.VirtualNetwork.ResourceGroup` to `s.ControlPlane.Spec.ResourceGroupName` in the AzureManagedControlPlane defaulting webhook and always use the `VirtualNetwork.ResourceGroup` value here?